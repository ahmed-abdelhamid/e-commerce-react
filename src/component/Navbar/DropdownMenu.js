import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { logout } from '../../action/auth/auth';

export const DropdownMenu = (props) => (
  <div className="navbar-nav nav-item dropdown">
    <span className="nav-link dropdown-toggle" id="navbarDropdown" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
      {props.fullname}
    </span>
    <div className="dropdown-menu" aria-labelledby="navbarDropdown">
      <Link to="/edit" className="dropdown-item">Edit Profile</Link>
      <Link to="/setting" className="dropdown-item">Setting</Link>
      <button
        className="dropdown-item"
        onClick={() => props.logout()}
      >
        Logout
      </button>
    </div>
  </div>
)

const mapStateToProps = ({ auth }) => ({
  fullname: `${auth.firstName} ${auth.lastName}`
});

const mapDispatchToProps = (dispatch) => ({
  logout: () => dispatch(logout())
});

export default connect(mapStateToProps, mapDispatchToProps)(DropdownMenu);