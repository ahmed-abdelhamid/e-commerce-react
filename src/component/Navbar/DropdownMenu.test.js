import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { DropdownMenu } from './DropdownMenu';

test('should render DropdownMenu correctly', () => {
  const fullname = 'Ahmed Abdelhamid';
  const wrapper = shallow(<DropdownMenu fullname={fullname} />);
  expect(toJSON(wrapper)).toMatchSnapshot();
});