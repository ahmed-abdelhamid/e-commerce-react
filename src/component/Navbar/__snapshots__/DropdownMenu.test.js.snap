// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render DropdownMenu correctly 1`] = `
<div
  className="navbar-nav nav-item dropdown"
>
  <span
    aria-expanded="false"
    aria-haspopup="true"
    className="nav-link dropdown-toggle"
    data-toggle="dropdown"
    id="navbarDropdown"
  >
    Ahmed Abdelhamid
  </span>
  <div
    aria-labelledby="navbarDropdown"
    className="dropdown-menu"
  >
    <Link
      className="dropdown-item"
      replace={false}
      to="/edit"
    >
      Edit Profile
    </Link>
    <Link
      className="dropdown-item"
      replace={false}
      to="/setting"
    >
      Setting
    </Link>
    <button
      className="dropdown-item"
      onClick={[Function]}
    >
      Logout
    </button>
  </div>
</div>
`;
